cluster/src/main/scala/org/apache/spark/memory/SnappyUnifiedMemoryManager.scala

```diff
@@ -543,8 +569,10 @@ class SnappyUnifiedMemoryManager private[memory](
       blockId: BlockId,
       numBytes: Long,
       memoryMode: MemoryMode,
-      shouldEvict: Boolean): Boolean = {
-    synchronized {
+      shouldEvict: Boolean): Boolean = synchronized {
+    // used for temporarily adjusting for ByteBuffer.allocateDirect usage
+    var storagePoolAdjustedSize = 0L
```

Comment:
What is this adjustment about?

Reply:
This is to account for Java ByteBuffer.allocateDirect() calls that don't go through UMM. If there are too many of these it can be a cause for memory consumption going much more than configured via memory-size+heap-size causing kernel's OOM killer to kick in. This will temporarily increment by reading those allocations from JDK's accounting by reflection (UnsafeHolder.getDirectReservedMemory) and pretending that actual pool size is smaller by this amount.

Reply:
ok.